python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -396,22 +401,59 @@ def launch_run(self, context: LaunchRunContext) -> None:
 
         # launchType and capacityProviderStrategy are incompatible - prefer the latter if it is set
         if "launchType" in run_task_kwargs and run_task_kwargs.get("capacityProviderStrategy"):
-            del run_task_kwargs["launchType"]
-
-        # Run a task using the same network configuration as this processes's task.
-        response = self.ecs.run_task(**run_task_kwargs)
+            del run_task_kwargs["launchType"]            
+
+        # function to reduce duplicate code and reduce nesting
+        def run_task():
+            response = self.ecs.run_task(**run_task_kwargs)
+            tasks = response["tasks"]
+            if not tasks:
+                failures = response["failures"]
+                exceptions = []
+                for failure in failures:
+                    arn = failure.get("arn")
+                    reason = failure.get("reason")
+                    detail = failure.get("detail")
+                    exceptions.append(Exception(f"Task {arn} failed because {reason}: {detail}"))
+                    if reason == "Capacity is unavailable at this time. Please try again later or in a different availability zone":
```

Comment:
this error can be hit even if you are not using fargate spot instances, and you might want to retry. Can we configure the number of retries to use if this happens separately from the capacity provider strategy?